src/uri.rs

```diff
@@ -0,0 +1,261 @@
+#![allow(dead_code)]
+
+use reqwest;
+
+use serde_derive::Deserialize;
+
+// these item structs should be namespaced into the v2 api.
+#[derive(Deserialize, Debug)]
+struct ItemAttribute {
```

Comment:
All of these Item types are not publicly accessible or used in non-test code.  If these are intended for testing then they should be moved to the testing module.  However, testing URI building does not really require fully testing the deserialization of the received data.

Reply:
Yeah, I can see moving them into the testing section. 
And yeah they aren't directly needed for testing uri building, but it certainly cooler, to actually see data be returned and deserialized, also helps with my understanding of how the libraries interact which each other.